sematic/db/queries.py

```diff
@@ -221,6 +222,39 @@ def save_run(run: Run) -> Run:
     return run
 
 
+def get_job(job_name: str, job_namespace: str) -> Optional[Job]:
+    """Get a job by name and namespace (or None if it doesn't exist)"""
+    with db().get_session() as session:
+        return (
+            session.query(Job)
+            .filter(Job.name == job_name)
+            .filter(Job.namespace == job_namespace)
+            .one_or_none()
+        )
+
+
+def save_job(job: Job) -> Job:
+    """Save a job to the db, updating an existing one if present."""
+    existing_job = get_job(job_name=job.name, job_namespace=job.namespace)
+    if existing_job is not None:
+        # do this to ensure that we are updating the history based on what's
+        # actually already in the DB.
+        existing_job.details = job.details
+        existing_job.update_status(job.latest_status)
+        job = existing_job
+    with db().get_session() as session:
+        session.merge(job)
+        session.commit()
+
+        return job
```

Comment:
Please do the read and update in a single session, in order to avoid a race condition.

Reply:
done